localstack/services/cloudformation/resource_providers/aws_cloudformation_macro.py

```diff
@@ -0,0 +1,101 @@
+# LocalStack Resource Provider Scaffolding v2
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional, TypedDict
+
+import localstack.services.cloudformation.provider_utils as util
+from localstack.services.cloudformation.resource_provider import (
+    OperationStatus,
+    ProgressEvent,
+    ResourceProvider,
+    ResourceRequest,
+)
+from localstack.services.cloudformation.stores import get_cloudformation_store
+
+
+class CloudFormationMacroProperties(TypedDict):
+    FunctionName: Optional[str]
+    Name: Optional[str]
+    Description: Optional[str]
+    Id: Optional[str]
+    LogGroupName: Optional[str]
+    LogRoleARN: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class CloudFormationMacroProvider(ResourceProvider[CloudFormationMacroProperties]):
+    TYPE = "AWS::CloudFormation::Macro"  # Autogenerated. Don't change
```

Comment:
Good work on implementing these. I wasn't aware of all the resources that CFn uses. Gonna have to dig in more into documentation